File: February2022/Java/SubarraySumEqualsK.java

class Solution {
	public int subarraySum(int[] nums, int k) {
		Map<Integer, Integer> map = new HashMap<>();
		map.put(0,1);
		int ans = 0;
		int sum = 0;
		for(int i =0;i<nums.length;i++){

			sum += nums[i];
			if(map.containsKey(sum- k)){
				ans += map.get(sum-k);
			}

			map.put(sum, map.getOrDefault(sum,0)+1);
		}
		return ans;
	}
}

File: February2022/Kotlin/SubarraySumEqualsK.kt
class Solution {
    fun subarraySum(nums: IntArray, k: Int): Int {
        val hs = HashMap<Int, Int>()
        hs[0] = 1
        var c = 0
        val ts = IntArray(nums.size)
        for (i in nums.indices) {
            ts[i] = nums[i]
            if (i > 0) {
                ts[i] += ts[i - 1]
            }
            if (hs.containsKey(ts[i] - k)) {
                c += hs[ts[i] - k]!!
            }
            hs[ts[i]] = hs.getOrDefault(ts[i], 0) + 1
        }
        return c
    }
}